internal_types/src/entry.rs

```diff
@@ -0,0 +1,1544 @@
+                    .fb
+                    .values_as_string_values()
+                    .expect("invalid flatbuffers")
+                    .values()
+                    .expect("flatbuffers StringValues must have string values set")
+                    .iter();
+
+                TypedValuesIterator::String(StringIterator {
+                    row_count: self.row_count,
+                    position: 0,
+                    null_mask: self.fb.null_mask(),
+                    values,
+                })
+            }
+            entry_fb::ColumnValues::I64Values => {
+                let values_iter = self
+                    .fb
+                    .values_as_i64values()
+                    .expect("invalid flatbuffers")
+                    .values()
+                    .unwrap_or_else(|| Vector::new(&[], 0))
+                    .iter();
+
+                TypedValuesIterator::I64(ValIterator {
+                    row_count: self.row_count,
+                    position: 0,
+                    null_mask: self.fb.null_mask(),
+                    values_iter,
+                })
+            }
+            entry_fb::ColumnValues::F64Values => {
+                let values_iter = self
+                    .fb
+                    .values_as_f64values()
+                    .expect("invalid flatbuffers")
+                    .values()
+                    .unwrap_or_else(|| Vector::new(&[], 0))
+                    .iter();
+
+                TypedValuesIterator::F64(ValIterator {
+                    row_count: self.row_count,
+                    position: 0,
+                    null_mask: self.fb.null_mask(),
+                    values_iter,
+                })
+            }
+            entry_fb::ColumnValues::U64Values => {
+                let values_iter = self
+                    .fb
+                    .values_as_u64values()
+                    .expect("invalid flatbuffers")
+                    .values()
+                    .unwrap_or_else(|| Vector::new(&[], 0))
+                    .iter();
+
+                TypedValuesIterator::U64(ValIterator {
+                    row_count: self.row_count,
+                    position: 0,
+                    null_mask: self.fb.null_mask(),
+                    values_iter,
+                })
+            }
+            entry_fb::ColumnValues::BytesValues => unimplemented!(),
+            _ => panic!("unknown fb values type"),
+        }
+    }
+}
+
+/// Wrapper for the iterators for the underlying column types.
+#[derive(Debug)]
+pub enum TypedValuesIterator<'a> {
+    Bool(BoolIterator<'a>),
+    I64(ValIterator<'a, i64>),
+    F64(ValIterator<'a, f64>),
+    U64(ValIterator<'a, u64>),
+    String(StringIterator<'a>),
+}
+
+impl<'a> TypedValuesIterator<'a> {
+    pub fn bool_values(self) -> Option<Vec<Option<bool>>> {
+        match self {
+            Self::Bool(b) => Some(b.collect::<Vec<_>>()),
+            _ => None,
+        }
+    }
```

Comment:
it seems a shame to go through all the trouble with flatbuffers to avoid a copy and then copy the data values here. 

I wonder if something like this could work ?

```suggestion
    pub fn bool_values(self) -> Option<impl Iterator<Item=Option<bool>>{
        match self {
            Self::Bool(b) => Some(b),
            _ => None,
        }
    }
```

Reply:
I actually just had this function as a convenience for tests. My intention was to use the iterator directly in the MB.